Multisig/UI/Transaction/TransactionDetailsViewController/TransactionDetailsViewController.swift

```diff
@@ -121,25 +121,59 @@ class TransactionDetailsViewController: LoadableViewController, UITableViewDataS
     }
 
     private var showsActionsViewContrainer: Bool  {
-        showsRejectButton || showConfirmButton
+        canSign && (showsRejectButton || showConfirmButton)
+    }
+
+    private var canSign: Bool {
+        if let signingKey = PrivateKeyController.signingKeyAddress,
+           let signingAddress = AddressString(signingKey),
+           case let SCGModels.TransactionDetails.DetailedExecutionInfo.multisig(multisigTx)? = tx?.detailedExecutionInfo,
+           multisigTx.isSigner(address: signingAddress) {
+            return true
+        }
+
+        return false
     }
 
     private var showsRejectButton: Bool {
         switch self.tx?.txInfo {
-
         case .rejection(_):
             return false
         default:
-            return tx?.multisigInfo != nil
+            guard let multisigInfo = tx?.multisigInfo,
+                  let status = tx?.txStatus
+                    else { return false }
+
+            if status == .awaitingExecution,
+                !multisigInfo.isRejected() {
```

Comment:
it returns false if status is awaiting execution and tx is not rejected - I think in this case we should show rejection button, but it returns false